src/tmpfiles/tmpfiles.c

```diff
@@ -1261,7 +1261,9 @@ static int create_item(Item *i) {
                 log_debug("%s directory \"%s\".", creation_mode_verb_to_string(creation), i->path);
 
                 if (IN_SET(i->type, CREATE_SUBVOLUME_NEW_QUOTA, CREATE_SUBVOLUME_INHERIT_QUOTA)) {
-                        r = btrfs_subvol_auto_qgroup(i->path, 0, i->type == CREATE_SUBVOLUME_NEW_QUOTA);
+                        r = path_check_fstype(i->path, BTRFS_SUPER_MAGIC);
+                        if (r == 1)
+                                 r = btrfs_subvol_auto_qgroup(i->path, 0, i->type == CREATE_SUBVOLUME_NEW_QUOTA);
```

Comment:
I was wrong about openat(fd, ".", ...).  It doesn't access anything mounted on "fd" - it always returns the same object as 'fd'.  open("/proc/self/fd/%i") does the same thing.
So path_set_perms() in tmpfiles.c won't trigger an automount - it will change permissions on the automount point (tested and confirmed)
mkdir_parents() calls is_dir() first.  This will detect a directory (without triggering the automount) and so won't make the directory.  So mkdir/chmod/chown won't trigger the automount.
Enhancing chase_symlinks() to fail on an automount should be easy.  I assume you are suggesting putting the test in parse_line() - only when globs aren't expected.  I have a patch for that and will push it.
It silently ignores any (non-glob) path which would pass through any part of an autofs filesystem.  I realised that not all directories in an autofs filesystem are necessarily automount points, and it is not possible to differentiate between that which are, and those which aren't.